src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+
+import usersFromServer from '../../api/users';
+import { getUserById } from '../../services/getUsers';
+import { getMaxNumber } from '../../services/getMaxNumber';
+import { Todo } from '../../types/todo';
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+  todos: Todo[];
+};
+
+export const TodoForm: React.FC<Props> = ({ onSubmit, todos }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handlTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handlUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setHasTitleError(false);
+    setHasUserIdError(false);
+  };
+
+  const handlSubmit = (event: React.FormEvent) => {
```

Comment:
[CHECKLIST - CODE KNOWLEDGE] - While handling form values, trigger submit logic using `onSubmit` prop of form, instead of `onClick` submit button. You're doing it correctly here, good job!